openssl-1.1.0h: Fix c_rehash perl errors

[No upstream tracking] -- https://github.com/openssl/openssl/issues/5772

dofile.pl: Revert only quote stuff that actually needs quoting

This wasn't a good solution, too many things depend on the quotes being
there consistently.

Upstream-Status: Backport [https://github.com/openssl/openssl/commit/00701e5ea84861b74d9d624f21a6b3fcb12e8acd]
bug: 5772
Signed-off-by: Andrej Valek <andrej.valek@siemens.com>

diff --git a/util/dofile.pl b/util/dofile.pl
index 955224df7d..b0e20681dd 100644
--- a/util/dofile.pl
+++ b/util/dofile.pl
@@ -99,9 +99,9 @@ package main;
 # This adds quotes (") around the given string, and escapes any $, @, \,
 # " and ' by prepending a \ to them.
 sub quotify1 {
-    my $s = my $orig = shift @_;
+    my $s = shift @_;
     $s =~ s/([\$\@\\"'])/\\$1/g;
-    $s ne $orig || $s =~ /\s/ ? '"'.$s.'"' : $s;
+    '"'.$s.'"';
 }
 
 # quotify_l LIST
